src/Common/src/Interop/Unix/System.Native/Interop.Read.cs

```diff
@@ -18,6 +19,6 @@ internal static partial class Sys
         /// Note - on fail. the position of the stream may change depending on the platform; consult man 2 read for more info
         /// </returns>
         [DllImport(Libraries.SystemNative, SetLastError = true)]
-        internal static unsafe extern int Read(int fd, byte* buffer, int count);
+        internal static unsafe extern int Read(SafeFileHandle fd, byte* buffer, int count);
```

Comment:
I think it's ok to have the two different Read declarations, one for SafeFileHandle and one for SafePipeHandle, but you might also consider just having one declaration that takes the base SafeHandle.